lib/EnsEMBL/REST/Model/ga4gh/Beacon.pm

```diff
@@ -970,6 +1068,127 @@ sub get_dataset_allele_response {
   return $ds_response;
 }
 
+sub get_vep_frequency {
+  my $vep_consequences = shift;
+
+  my @gnomad_frequency;
+
+  foreach my $consequence (@{$vep_consequences}) {
+    foreach my $variant (@{$consequence->{colocated_variants}}) {
+      foreach my $allele (keys %{$variant->{frequencies}}) {
+        my $frequencies = $variant->{frequencies}->{$allele};
+        foreach my $key (keys %{$frequencies}) {
+          if($key =~ /gnomad/) {
+            my ($gnomad_type, $pop) = $key =~ /(.*)_(.*)/;
+            my $freq_obj;
+            $freq_obj->{alleleFrequency} = $frequencies->{$key};
+            $freq_obj->{allele} = $allele;
+
+            if($gnomad_type) {
+              $pop = $gnomad_pop->{$pop} ? $gnomad_pop->{$pop} : $pop;
+              $gnomad_type =~ s/gnomade/gnomAD exomes/;
+              $gnomad_type =~ s/gnomadg/gnomAD genomes/;
+              $freq_obj->{population} = $gnomad_type . ' ' .$pop;
+            }
+            else {
+              $key =~ s/gnomade/gnomAD exomes/;
+              $key =~ s/gnomadg/gnomAD genomes/;
+              $freq_obj->{population} = $key;
+            }
+
+            push @gnomad_frequency, $freq_obj;
+          }
+        }
+      }
+    }
+  }
+
+  return \@gnomad_frequency;
+}
+
+# Input: vep consequences list
+# Output: hash of molecular interationc and list of gene ontologies
+sub get_vep_molecular_attribs {
+  my $vep_consequences = shift;
+
+  my @gene_ontology;
+  my %molecular_interactions;
+  my @intact_data;
+  my @phenotypes;
+  my %unique_disgenet;
+  my @disgenet_data;
+  my @cadd_scores;
+  my %cadd_unique;
+
+  my %results;
+
+  foreach my $consequence (@{$vep_consequences}) {
+    foreach my $transcript_consequences (@{$consequence->{'transcript_consequences'}}) {
+      # IntAct
+      foreach my $intact (@{$transcript_consequences->{'intact'}}) {
+        push @intact_data, $intact;
+      }
+      $molecular_interactions{'IntAct'} = \@intact_data if(scalar @intact_data > 0);
+
+      # GO
+      if($transcript_consequences->{'go'}) {
+        my @go_terms = split ',', $transcript_consequences->{'go'};
+        foreach my $go (@go_terms) {
+          my ($id, $term) = $go =~ /(.*):(.*)/;
+          my $cons;
+          $cons->{id} = $id;
+          $cons->{label} = $term;
+          push @gene_ontology, $cons;
+        }
+      }
+
+      # DisGeNET
+      if($transcript_consequences->{'disgenet'}) {
+        foreach my $disgenet (@{$transcript_consequences->{'disgenet'}}) {
+          my $key = $disgenet->{score} . '-' . $disgenet->{pmid} . '-' . $disgenet->{diseaseName};
+          if(!$unique_disgenet{$key}) {
+            my $disgenet_obj;
+            $disgenet_obj->{annotatedWith}->{toolName} = 'DisGeNET';
+            $disgenet_obj->{annotatedWith}->{version} = 'v7';
+            $disgenet_obj->{conditionId} = $disgenet->{diseaseName};
+            $disgenet_obj->{evidenceType} = 'PMID:' . $disgenet->{pmid};
```

Comment:
`evidenceType` should have have an `id` and a `label`. They also recommend using the [Evidence & Conclusion Ontology (ECO)](https://www.ebi.ac.uk/ols/ontologies/eco) terms: http://docs.genomebeacons.org/schemas-md/obj/evidenceType/

For a PubMed reference, I think the `evidenceType` should have either:
- `id` [IAO:000011](https://www.ebi.ac.uk/ols/ontologies/eco/properties?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FIAO_0000119&lang=en&viewMode=All&siblings=false) and PMID ID as `label`
- `id` [isDefinedBy](https://www.ebi.ac.uk/ols/ontologies/eco/properties?iri=http%3A%2F%2Fwww.w3.org%2F2000%2F01%2Frdf-schema%23isDefinedBy) and PMID ID as `label`